normandy/control/static/control/js/components/RecipeForm.js

```diff
@@ -153,10 +154,37 @@ export default composeRecipeContainer(reduxForm({
     ? props.location.state.selectedRevision
     : null;
 
+  const formatErrors = payload => {
+    let errors = payload;
+
+    if (_.isObject(payload)) {
+      const invalidFields = Object.keys(payload);
+      if (invalidFields.length > 0) {
+        errors = isNaN(invalidFields[0]) ? {} : [];
```

Comment:
That's a pretty confusing incantation. What's going on here?

Overall, it might be a good idea to add a comment to this function that describes the kind of data it is expecting.